packages/order-utils/src/asset_data_utils.ts

```diff
@@ -13,34 +23,25 @@ export const assetDataUtils = {
      * @return The hex encoded assetData string
      */
     encodeERC20AssetData(tokenAddress: string): string {
-        return ethUtil.bufferToHex(ethAbi.simpleEncode('ERC20Token(address)', tokenAddress));
+        const abiEncoder = new AbiEncoder.Method(constants.ERC20_METHOD_ABI as MethodAbi);
+        const args = [tokenAddress];
+        const assetData = abiEncoder.encode(args, encodingRules);
+        return assetData;
     },
     /**
      * Decodes an ERC20 assetData hex string into it's corresponding ERC20 tokenAddress & assetProxyId
      * @param assetData Hex encoded assetData string to decode
      * @return An object containing the decoded tokenAddress & assetProxyId
      */
     decodeERC20AssetData(assetData: string): ERC20AssetData {
-        const data = ethUtil.toBuffer(assetData);
-        if (data.byteLength < constants.ERC20_ASSET_DATA_BYTE_LENGTH) {
-            throw new Error(
-                `Could not decode ERC20 Proxy Data. Expected length of encoded data to be at least ${
-                    constants.ERC20_ASSET_DATA_BYTE_LENGTH
-                }. Got ${data.byteLength}`,
-            );
-        }
-        const assetProxyId = ethUtil.bufferToHex(data.slice(0, constants.SELECTOR_LENGTH));
-        if (assetProxyId !== AssetProxyId.ERC20) {
-            throw new Error(
-                `Could not decode ERC20 Proxy Data. Expected Asset Proxy Id to be ERC20 (${
-                    AssetProxyId.ERC20
-                }), but got ${assetProxyId}`,
-            );
-        }
-        const [tokenAddress] = ethAbi.rawDecode(['address'], data.slice(constants.SELECTOR_LENGTH));
+        assetDataUtils.assertIsERC20AssetData(assetData);
+        const assetProxyId = assetDataUtils.decodeAssetProxyId(assetData);
+        const abiEncoder = new AbiEncoder.Method(constants.ERC20_METHOD_ABI as MethodAbi);
```

Comment:
Should just declare the type as `MethodAbi` inside of `constants` in order to avoid all these type assertions?